distributed/shuffle/tests/test_shuffle.py

```diff
@@ -2207,3 +2207,65 @@ def test_sort_values_p2p_with_existing_divisions(client):
             check_index=False,
             sort_results=False,
         )
+
+
+class BlockedBarrierShuffleRun(DataFrameShuffleRun):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.in_barrier = asyncio.Event()
+        self.block_barrier = asyncio.Event()
+
+    async def barrier(self):
+        self.in_barrier.set()
+        await self.block_barrier.wait()
+        return await super().barrier()
+
+
+@mock.patch(
+    "distributed.shuffle._shuffle.DataFrameShuffleRun",
+    BlockedBarrierShuffleRun,
+)
+@gen_cluster(client=True, nthreads=[("", 1)] * 2)
+async def test_unpack_gets_rescheduled_on_bystanding_worker(c, s, a, b):
+    """Test that when we lost annotations, unpack tasks run on bystanding workers
+    do not fail but get rescheduled.
+
+    This test is heavily using worker restrictions to simulate certain scheduler
+    decisions of placing tasks.
+    """
+    await invoke_annotation_chaos(1.0, c)
+
+    expected = pd.DataFrame({"a": list(range(10))})
+    ddf = dd.from_pandas(expected, npartitions=2)
+
+    # B should not participate in the initial shuffle phases, so place all outputs on A
+    def mock_get_worker_for_range_sharding(
+        output_partition: int, workers: list[str], npartitions: int
+    ) -> str:
+        return a.address
```

Comment:
This could also be established by starting a new worker after the shuffle started, couldn't it?

Reply:
Good point, that should also work. I'll see if I can simplify the test with that.

Reply:
I've simplified the test a bit.